src/sync.rs

```diff
@@ -418,14 +418,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.machine.threads.sync.condvars[id].waiters.retain(|waiter| waiter.thread != thread);
     }
 
-    fn futex_wait(&mut self, addr: Pointer<stacked_borrows::Tag>, thread: ThreadId) {
+    fn futex_wait(&mut self, addr: Pointer, thread: ThreadId) {
```

Comment:
From an API perspective, I think it is better to do the tag erasure here inside `futex_wait` than to make the caller do it. Same for `futex_wake`.